Source/missiles.cpp

```diff
@@ -865,7 +861,7 @@ bool PlayerMHit(int pnum, int m, int dist, int mind, int maxd, int mtype, bool s
 				plr[pnum]._pHPBase = plr[pnum]._pMaxHPBase;
 			}
 
-			if (plr[pnum]._pHitPoints >> 6 <= 0) {
+			if (plr[pnum]._pHitPoints / 64 <= 0) {

```

Comment:
The low 6 bits store the "fractional" part

Reply:
I've reverted all changes from this PR related to HP and Mana.